### PR TITLE
Henry's Upgrade

### PR DESCRIPTION
**Pull request by mengfoobar**
Completed challenge as specified on the instructions.

Cheers!

### CODE REVIEW

File: app/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+        window.addEventListener("resize",()=>{
+            this.setState({
+                mobile: screen.width <600
+            })
+        } );
+
+        server.on('load', (todos) => {
+            this.setState({
+                todos:todos
```
Comment (NicHaley) [inline app/components/app.jsx]: ES6 shorthand -> this.setState({ todos })

File: app/components/app.jsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+        server.on('undoRemove', () => {
+            this.undoRemove()
+        });
+
+        server.on('setCheck', (index) => {
+            this.setCheck(index)
+        });
+
+        server.on('disconnect', ()=>{
```
Comment (NicHaley) [inline app/components/app.jsx]: nice!

File: app/components/app.jsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    add(todo=null){
+        const {todoInput, todos} = this.state;
+
+        const todoToAppend = todo || (todoInput ? new Todo(todoInput): null);
+
+        if(!todoToAppend){
+            return;
+        }
+
+        const curTodos =  List(todos);
```
Comment (NicHaley) [inline app/components/app.jsx]: Nice use of Immutable 👍

File: app/components/app.jsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            })
+        });
+    }
+
+    handleTodoInputChange(e){
+        this.setState({todoInput: e.target.value});
+    }
+
+    add(todo=null){
+        const {todoInput, todos} = this.state;
```
Comment (NicHaley) [inline app/components/app.jsx]: Good code style

File: app/components/app.jsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+                    </button>
+
+                </div>
+
+                <ul id="todo-list" className={style.todos}>
+                    {todos.map((t, i)=>{
+                        return (
+                            <TodoItem key={i} todo={t} index={i}
+                                      setCheck={()=>{
+                                          cacheMode ? this.setCheck(i) : server.emit('setCheck', i)
```
Comment (NicHaley) [inline app/components/app.jsx]: nice solution for handling disconnects

File: app/components/app.jsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+                cacheMode:false
+            })
+        });
+    }
+
+    handleTodoInputChange(e){
+        this.setState({todoInput: e.target.value});
+    }
+
+    add(todo=null){
```
Comment (NicHaley) [inline app/components/app.jsx]: It would have been beneficial to clear the input after adding a todo

File: app/components/app.jsx [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+    }
+
+    render() {
+        const {todoInput, todos, mobile, cacheMode} = this.state;
+        const {mInputBox,mBtn} = style;
+        const btnClass = `${mobile? mBtn:""}`
+
+        return (
+            <div>
+                <input id="todo-input" type="text" placeholder="Feed the cat" value={todoInput}
```
Comment (NicHaley) [inline app/components/app.jsx]: Would have been nice to handle adding a todo on Enter with `onKeyPress`

**Comment by mengfoobar** [comment]
@NicHaley thanks so much for the code review! really appreciate your time and valuable feedback.
I fixed the suggestions you proposed  :)  Totally biffed it on the complete all one..., just added it on :+1: 

As for additional styles, I didn't add any more styling other than making it mobile responsive, because by "design change", I thought it meant adding an additional functional design (which is where the undo feature came from)  （´∀｀）

Again, thank you so much, always glad to get constructive criticism from a seasoned dev!

File: app/components/app.jsx (hunk 7 above)
Comment (mengfoobar) [inline app/components/app.jsx]: agreed! just added it

File: app/components/app.jsx (hunk 6 above)
Comment (mengfoobar) [inline app/components/app.jsx]: added :)

Comment (NicHaley) [inline app/components/app.jsx]: nice!

**Comment by NicHaley** [comment]
Np Henry!

File: app/components/app.jsx [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+        this.setState({
+            todos: newTodosArr
+        }, ()=>{server.emit('setDB', newTodosArr)})
+
+    }
+
+    render() {
+        const {todoInput, todos, mobile, cacheMode} = this.state;
+        const {mInputBox,mBtn} = style;
+        const btnClass = `${mobile? mBtn:""}`
```
Comment (NicHaley) [inline app/components/app.jsx]: Would have been better to handle mobile styles with CSS media queries instead of with javascript

**Pull request closed without merging**